packages/next-swc/crates/next-api/src/project.rs

```diff
@@ -254,6 +282,19 @@ impl Project {
         self.project_fs().root()
     }
 
+    /// Returns a path to dist_dir resolved from project root path.
+    /// Currently this is used to embed process.env.__NEXT_DIST_DIR.
+    /// [Note] in webpack-config, it is being injected when
+    /// dev && (isClient || isEdgeServer)
+    #[turbo_tasks::function]
+    async fn dist_root_path(self: Vc<Self>) -> Result<Vc<FileSystemPath>> {
+        let this = self.await?;
+        Ok(self
+            .project_root_path()
+            .root()
+            .join(this.dist_dir.to_string()))
+    }
```

Comment:
Do we want to support reading from this disk path? Should this just be a `Vc<String>` instead?

Reply:
I think Vc<String> would work, this is mainly to set process.env so next-dev's overlay / reloader can refer, turbopack itself may not need to reference this path by itself.